SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -49,6 +57,31 @@ FLocalDeploymentManager::FLocalDeploymentManager()
 #endif
 }
 
+bool FLocalDeploymentManager::PreRunChecks()
+{
+	FString SpatialExistenceCheckResult;
+	int32 ExitCode;
+	FSpatialGDKServicesModule::ExecuteAndReadOutput(SpatialExe, TEXT("version"), FSpatialGDKServicesModule::GetSpatialOSDirectory(), SpatialExistenceCheckResult, ExitCode);
+
+	if (ExitCode != 0)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Warning, TEXT("Spatial.exe does not exist on this machine! Please make sure Spatial is installed before trying to start a local deployment. %s"), *SpatialExistenceCheckResult);
+		return false;
+	}
+
+	FString SpotExistenceCheckResult;
+	FString StdErr;
+	FPlatformProcess::ExecProcess(*GetSpotExe(), TEXT("version"), &ExitCode, &SpotExistenceCheckResult, &StdErr);
```

Comment:
Nit: it might be nice to have GetSpotExe() call in a separate line and make sure that it doesn't return an empty string (but as long as GetSpotExe() function is not changed - that should never happen)

Reply:
This is how we do it in the rest of this class, I think it's fine, if that method ever returns empty string then everything is going to fall over anyway.